src/main/java/de/tum/cit/aet/artemis/core/service/telemetry/EurekaClientService.java

```diff
@@ -0,0 +1,96 @@
+package de.tum.cit.aet.artemis.core.service.telemetry;
+
+import static de.tum.cit.aet.artemis.core.config.Constants.PROFILE_SCHEDULING;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+import java.util.Collections;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Profile;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestClientException;
+import org.springframework.web.client.RestTemplate;
+
+import com.fasterxml.jackson.core.JacksonException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+@Profile(PROFILE_SCHEDULING)
+@Service
+public class EurekaClientService {
+
+    private static final Logger log = LoggerFactory.getLogger(EurekaClientService.class);
+
+    @Value("${eureka.client.service-url.defaultZone}")
+    private String eurekaServiceUrl;
+
+    private final RestTemplate restTemplate;
+
+    public EurekaClientService(RestTemplate restTemplate) {
+        this.restTemplate = restTemplate;
+    }
+
+    /**
+     * Retrieves the number of Artemis application instances registered with the Eureka server.
+     * <p>
+     * This method makes an HTTP GET request to the Eureka server's `/api/eureka/applications` endpoint to
+     * retrieve a list of all registered applications. It then filters out the Artemis application
+     * and returns the number of instances associated with it. If any error occurs during the request
+     * (e.g., network issues, parsing issues, invalid URI), the method returns 1 as the default value.
+     *
+     * @return the number of Artemis application instances, or 1 if an error occurs.
+     */
+    public long getNumberOfReplicas() {
```

Comment:
Would this be the same as something like `eurekaClient.getApplications(eurekaServiceUrl).getInstancesByVirtualHostName("ARTEMIS").size()`?

https://javadoc.io/doc/com.netflix.eureka/eureka-client/latest/com/netflix/discovery/EurekaClient.html

- assuming the `EurekaClient` is available here via injection
- Is `ARTEMIS` the same as `${spring.application.name}`, or is there another config value where this can be changed?

Reply:
I tried this, but unfortunately I didn't find a way to inject it here. Though, for now I completely removed this anyway, as the approach to Thread sleep until (hopefully) all other instances are registered is really not great. For now there'll only be a flag, if it's multi or single node.